lib/roast/workflow/base_workflow.rb

```diff
@@ -113,7 +113,8 @@ def chat_completion(**kwargs)
         log_and_raise_error(error, message, step_model || model, kwargs, execution_time)
       rescue => e
         execution_time = Time.now - start_time
-        log_and_raise_error(e, e.message, step_model || model, kwargs, execution_time)
+        enhanced_message = build_error_message(e)
+        log_and_raise_error(e, enhanced_message, step_model || model, kwargs, execution_time)
```

Comment:
```suggestion
        log_and_raise_error(e, enhanced_message(e), step_model || model, kwargs, execution_time)
```

Maybe makes more sense to call the method "enhanced_message(error)"?